src/java/org/apache/cassandra/index/sai/disk/oldlucene/LegacyByteBuffersDataInput.java

```diff
@@ -0,0 +1,359 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.index.sai.disk.oldlucene;
+
+import java.io.EOFException;
+import java.io.IOException;
+import java.nio.BufferUnderflowException;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.apache.lucene.store.DataInput;
+import org.apache.lucene.store.RandomAccessInput;
+import org.apache.lucene.util.Accountable;
+import org.apache.lucene.util.RamUsageEstimator;
+
+/**
+ * A {@link DataInput} implementing {@link RandomAccessInput} and reading data from a
+ * list of {@link ByteBuffer}s. This uses the big-endian byte ordering of Lucene 7.5.
+ * Note that this participates in the type hierarchy of the modern Lucene depencency,
+ * so DataInput methods that assume little-endianness must be overridden.
+ */
+public final class LegacyByteBuffersDataInput extends DataInput implements Accountable, RandomAccessInput {
```

Comment:
Is this and the other LegacyByteBuffersXXX classes copied and adapted from Lucene, or have you written them from scratch?

Reply:
Ah, so that explains why the code formatting is different than in the rest of the project. I wonder, if we already modified them to adapt to our codebase, shouldn't we also reformat them so they match our project code style? Especially the braces? Otherwise, if we continue editing, it will be easy to end up with a mixture of styles (people have code style set in the IDE to match C* code style).

Reply:
On the other hand, reformatting might cause conflicts when porting newer canges from lucene... Although, that won't happen, will it?